Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.cpp

```diff
@@ -48,30 +46,25 @@
 
 namespace WebKit {
 
-RemoteGPU::RemoteGPU(WebGPUIdentifier identifier, GPUConnectionToWebProcess& gpuConnectionToWebProcess, RemoteRenderingBackend& renderingBackend, IPC::StreamServerConnection::Handle&& connectionHandle)
-    : m_gpuConnectionToWebProcess(gpuConnectionToWebProcess)
-    , m_workQueue(IPC::StreamConnectionWorkQueue::create("WebGPU work queue"))
+RemoteGPU::RemoteGPU(WebGPUIdentifier identifier, IPC::StreamServerConnection::Handle&& connectionHandle)
+    : m_workQueue(IPC::StreamConnectionWorkQueue::create("WebGPU work queue"))
     , m_streamConnection(IPC::StreamServerConnection::create(WTFMove(connectionHandle), workQueue()))
     , m_objectHeap(WebGPU::ObjectHeap::create())
     , m_identifier(identifier)
-    , m_renderingBackend(renderingBackend)
-    , m_webProcessIdentifier(gpuConnectionToWebProcess.webProcessIdentifier())
 {
 }
 
 RemoteGPU::~RemoteGPU() = default;
 
 void RemoteGPU::initialize()
 {
-    assertIsMainRunLoop();
```

Comment:
why is this assertion and the below one removed?

Reply:
Because the message is now being received on the stream connection, which has its own message queue. GPUConnectionToWebProcess messages are received on the main thread, but stream connection messages are received on a message queue